### ISSUE TITLE
[BUG] call_asgi doesn't respect base_url

### ISSUE THREAD

**Issue by jswk** [schemathesis/schemathesis#1366]
**Checklist**

- [x] I checked the [FAQ section](https://schemathesis.readthedocs.io/en/stable/faq.html#frequently-asked-questions) of the documentation
- [x] I looked for similar issues in the [issue tracker](https://github.com/schemathesis/schemathesis/issues)

**Describe the bug**

At least for the `call_asgi`, the base path is ignored despite setting a proper `server` key in yaml or passing `base_url`.

**To Reproduce**
Steps to reproduce the behavior, schemathesis will generate examples for `/foo`, instead of `/v1/foo`:
```python
import os
import schemathesis
from fastapi import FastAPI

app = FastAPI()

@app.get("/v1/foo")
def read_root():
    return {"Hello": "World"}

schema = schemathesis.from_path(
    os.path.dirname(__file__) + "/bug.yaml",
#    base_url="http://localhost/v1" # uncommenting this doens't help either
)

@schema.parametrize()
def test_api(case):
    response = case.call_asgi(app)
    case.validate_response(response)
```

The minimal schema that causes the bug, placed under `bug.yaml` for the above to work.
```yaml
openapi: 3.0.3
info:
  version: 0.0.1
  title: foo
servers:
  - url: https://example.org/v1
paths:
  /foo:
    get:
      responses:
        '200':
          description: OK
```

**Expected behavior**
In the above case, schemathesis will generate examples for `/v1/foo`.

**Environment (please complete the following information):**
 - OS: Ubuntu 20.04
 - Python version: 3.10.0
 - Schemathesis version: 3.12.0
 - Spec version: Open API 3.0.3

**Comment by Stranger6667** [comment]
Hi @jswk 

Thanks for reporting and providing clean reproduction steps! 
The problem is that `Case.call_asgi` has `http://testserver` as the default value for `base_url`. Though, generated code samples have the right path. As a temporary workaround:

```python
@schema.parametrize()
def test_api(case):
    response = case.call_asgi(app, base_url="http://testserver/v1")
    case.validate_response(response)
```

I'll work on a fix

**Comment by Stranger6667** [comment]
Fixed in `3.12.1`! Thanks once again for reporting it :)

**Comment by jswk** [comment]
Great! Thanks for a swift response!
